lib/Browser.php

```diff
@@ -35,10 +36,10 @@ class Browser implements BuzzClientInterface
     /** @var ResponseInterface */
     private $lastResponse;
 
-    public function __construct(BuzzClientInterface $client = null)
+    public function __construct(BuzzClientInterface $client = null, ?ResponseFactoryInterface $factory = null)
```

Comment:
Hm... This factory is used for both the request and response.

Reply:
One idea is to split them up into two variables.

Reply:
What I could find it's only used once on line [254](https://github.com/kriswallsmith/Buzz/blob/master/lib/Browser.php#L254)

And it's docblock is type hinted as [RequestFactory](https://github.com/kriswallsmith/Buzz/blob/master/lib/Browser.php#L24)

Reply:
Sorry, you are correct. But this should also be passed to the `new FileGetContentsClient`, right?